src/librustc_mir/transform/qualify_min_const_fn.rs

```diff
@@ -326,7 +326,7 @@ fn check_terminator(
                     abi::Abi::Rust if tcx.is_min_const_fn(def_id) => {},
                     abi::Abi::Rust => return Err((
                         span,
-                        "can only call other `min_const_fn` within a `min_const_fn`".into(),
+                        "can only call other `const` within a `const`".into(),
```

Comment:
Something like ```can only call another `const fn` within a `const` block``` sounds more natural.

Reply:
Neither this PR nor the suggestion is accurate. This is *not* a const block (and with https://github.com/rust-lang/rfcs/pull/2632 it will not always behave in a similar fashion).

Moreover, I think we lose out on the difference between *stable* `const fn`s and unstable ones. This is the main point of `min_const_fn`s... I think we should check whether the current compiler can use unstable features and tweak the message based on that. E.g. a stable compiler can just talk about `const fn`s but the nightly compiler should emphasize the difference between stable and unstable ones.

Reply:
Maybe change it to

```
can only call other stable `const fn` within a stable `const fn`
```

?

Reply:
Ah, in that case we should make the error more explicit and add a note to the unstable const fn pointing out that it is undertake.

Reply:
That is, it's not clear whether it's the containing function or the called function that is the problem (or both).

Reply:
The problem is to be used same wording without distinction, I think. Should we try to work on this?

Reply:
kinda both. I'm not sure how easy it is to point to the unstable `const fn`, since it might be in another crate. We can just use a more verbose wording like

> can only call other `const fn` within a `const fn`, but `insert_fn_name_here` is not stable as `const fn`

Reply:
@oli-obk I changed wording.